crates/wascc-provider/src/states/terminated.rs

```diff
@@ -7,16 +7,15 @@ pub struct Terminated;
 
 #[async_trait::async_trait]
 impl State<PodState> for Terminated {
-    async fn next(
-        self: Box<Self>,
-        pod_state: &mut PodState,
-        _pod: &Pod,
-    ) -> anyhow::Result<Transition<PodState>> {
+    async fn next(self: Box<Self>, pod_state: &mut PodState, _pod: &Pod) -> Transition<PodState> {
         let mut lock = pod_state.shared.handles.write().await;
         if let Some(handle) = lock.get_mut(&pod_state.key) {
-            handle.stop().await?;
+            let stop_result = handle.stop().await;
+            if let Err(e) = stop_result {
+                return Transition::Fatal(e);
```

Comment:
Nit: I _think_ fatal works here, but I also wonder if we should transition back into `Terminated` again to retry. Trying to think of all the cases where an error would occur here and if it is recoverable (i.e. if stop fails, will it succeed if we try again?)

Reply:
Yeah, I wasn't sure.  But this is the current behaviour and after some unfortunate experiences trying to make more things into state transitions I decided to keep it for now.  It would be good to review all fatal/complete transitions, clarify the expected/desired behaviour, and ensure that the tests are checking for that behaviour.

(I doubt this particular case affected the tests but I definitely had one where changing to an error transition resulted in a pod not being terminated properly!)

Reply:
Yeah, that is fine for now. Just a nit as I was reasoning through things